WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListSiteView.swift

```diff
@@ -10,17 +10,31 @@ struct BlogListSiteView: View {
                 .frame(width: 40, height: 40)
 
             VStack(alignment: .leading) {
-                Text(site.title)
-                    .font(.callout.weight(.medium))
-                    .lineLimit(2)
+                HStack(alignment: .center) {
+                    Text(site.title)
+                        .font(.callout.weight(.medium))
 
+                    if let badge = site.badge {
+                        makeBadge(with: badge)
+                    }
+                }
                 Text(site.domain)
                     .font(.footnote)
                     .foregroundStyle(.secondary)
-                    .lineLimit(1)
             }
+            .lineLimit(1)
         }
     }
+
+    func makeBadge(with viewModel: BlogListSiteViewModel.Badge) -> some View {
+        Text(viewModel.title.uppercased())
```

Comment:
I think we'd get a lot of mileage out of a `Badge` component – worth it to do now?

Reply:
I'm not sure we are going to need this exact variant anywhere else. I'd suggest extracting if it becomes needed in more than one place.